src/components/atoms/Alert/Alert.tsx

```diff
@@ -68,12 +69,33 @@ const Alert: FC<AlertProps> = ({
   ...rest
 }) => {
   const theme = useThemeContext();
-  const { component: Icon, text } = theme.alert[severity];
+  // eslint-disable-next-line
+  const { component: IconComponent, text, FAVariant, customVariant } = theme.alert[severity];
+  let iconDefinition, CustomIcon;
+
+  if (FAVariant) {
+    const iconLookup: IconLookup = { prefix: FAVariant.prefix, iconName: FAVariant.iconName };
+    iconDefinition = findIconDefinition(iconLookup);
+
+    if (!iconDefinition) {
+      throw new Error(`Icon ${FAVariant.prefix}-${FAVariant.iconName} not found`);
+    }
+  }
+
+  if (customVariant) {
+    if (customVariant.iconName === 'exclamation') {
+      CustomIcon = <ExclamationIcon color={customVariant.color} />;
+    } else {
+      throw new Error(`Unknown custom icon name: ${customVariant.iconName}`);
+    }
+  }
 
   return (
     <Wrapper severity={severity} inline={inline} hasRoundedCorners={hasRoundedCorners} {...rest} theme={theme}>
       <IconWrapper severity={severity} theme={theme}>
-        <Icon />
+        {IconComponent && <IconComponent />}
```

Comment:
Use the ternaries for these, a bit more verbose but safer: https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx